homeassistant/components/braviatv/media_player.py

```diff
@@ -60,115 +63,70 @@
+                    DOMAIN,
+                    context={"source": SOURCE_IMPORT},
+                    data={CONF_HOST: host, CONF_PIN: pin},
+                )
+            )
 
-    if pin is None:
-        request_configuration(config, hass, add_entities)
-        return
 
-    # If we came here and configuring this host, mark as done
-    if host in _CONFIGURING:
-        request_id = _CONFIGURING.pop(host)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
-        _LOGGER.info("Discovery configuration done")
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add BraviaTV entities from a config_entry."""
+    ignored_sources = []
+    pin = config_entry.data[CONF_PIN]
+    braviarc = hass.data[DOMAIN][config_entry.entry_id]
 
-    braviarc = BraviaRC(host)
-    if not braviarc.connect(pin, CLIENTID_PREFIX, NICKNAME):
-        _LOGGER.error("Cannot connect to %s", host)
-        return
     try:
-        system_info = braviarc.get_system_info()
+        system_info = await hass.async_add_executor_job(braviarc.get_system_info)
     except TypeError:
-        _LOGGER.error("Cannot retrieve system info from %s", host)
-        return
-    mac = format_mac(system_info["macAddr"])
-    unique_id = system_info["cid"].lower()
-
-    # Save config
-    save_json(
-        hass.config.path(BRAVIA_CONFIG_FILE),
-        {host: {"pin": pin, "host": host, "mac": mac}},
-    )
-
-    add_entities([BraviaTVDevice(braviarc, name, pin, unique_id)])
-
-
-def request_configuration(config, hass, add_entities):
-    """Request configuration steps from the user."""
-    host = config[CONF_HOST]
-    name = config[CONF_NAME]
-
-    configurator = hass.components.configurator
-
-    # We got an error if this method is called while we are configuring
-    if host in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[host], "Failed to register, please try again."
-        )
+        _LOGGER.error("Cannot retrieve system info from TV")
         return
-
-    def bravia_configuration_callback(data):
-        """Handle the entry of user PIN."""
-
-        pin = data.get("pin")
-        _braviarc = BraviaRC(host)
-        _braviarc.connect(pin, CLIENTID_PREFIX, NICKNAME)
-        if _braviarc.is_connected():
-            setup_bravia(config, pin, hass, add_entities)
-        else:
-            request_configuration(config, hass, add_entities)
-
-    _CONFIGURING[host] = configurator.request_config(
-        name,
-        bravia_configuration_callback,
-        description=(
-            "Enter the Pin shown on your Sony Bravia TV."
-            "If no Pin is shown, enter 0000 to let TV show you a Pin."
-        ),
-        description_image="/static/images/smart-tv.png",
-        submit_caption="Confirm",
-        fields=[{"id": "pin", "name": "Enter the pin", "type": ""}],
+    braviarc._mac = system_info["macAddr"].lower()  # pylint: disable=protected-access
```

Comment:
Why is setting the mac address necessary?

Reply:
The MAC address is necessary to turn on the TV via Wake on LAN.

Reply:
If we need to set the mac address we should create a new instance and pass the mac address as the second parameter. The best would be if the library would handle this for us itself since it should already know the mac address from the system_info.

Reply:
Maybe we can store the mac address from the system info call in the config flow. Then we can pass the mac address from the beginning here.

Reply:
It would be nice if we didn't need to do another system info call here.